velox/exec/tests/ExpandTest.cpp

```diff
@@ -0,0 +1,159 @@
+              {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+          .project({"k1", "k2", "count_1", "sum_a", "max_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY GROUPING SETS ((k1), (k1), (k2))");
+}
+
+TEST_F(ExpandTest, cube) {
+  auto data = makeRowVectorData(1'000);
+
+  createDuckDbTable({data});
+
+  // Cube.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand({
+              {"k1", "k2", "a", "b", "0 as group_id_0"},
+              {"k1", "null", "a", "b", "1"},
+              {"null", "k2", "a", "b", "2"},
+              {"null", "null", "a", "b", "3"},
+          })
+          .singleAggregation(
+              {"k1", "k2", "group_id_0"},
+              {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+          .project({"k1", "k2", "count_1", "sum_a", "max_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY CUBE (k1, k2)");
+}
+
+TEST_F(ExpandTest, rollup) {
+  auto data = makeRowVectorData(1'000);
+
+  createDuckDbTable({data});
+
+  // Rollup.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand(
+              {{"k1 as foo", "k2", "a", "b", "0 as group_id_0"},
+               {"k1", "null", "a", "b", "1"},
+               {"null", "null", "a", "b", "2"}})
+          .singleAggregation(
+              {"foo", "k2", "group_id_0"},
+              {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+          .project({"foo", "k2", "count_1", "sum_a", "max_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY ROLLUP (k1, k2)");
+}
+
+TEST_F(ExpandTest, countDistinct) {
+  auto data = makeRowVectorData(1'000);
+
+  createDuckDbTable({data});
+
+  // count distinct.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand(
+              {{"k1 as foo", "k2", "a", "b", "0 as group_id_0"},
+               {"k1", "null", "a", "b", "1"},
+               {"null", "null", "a", "b", "2"}})
+          .singleAggregation(
+              {"foo", "k2", "group_id_0"},
+              {"count(distinct a) as count_a", "count(distinct b) as count_b"})
+          .project({"count_a", "count_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT count(distinct a), count(distinct b) FROM tmp GROUP BY ROLLUP (k1, k2)");
+}
+
+TEST_F(ExpandTest, invalidUseCases) {
+  auto data = makeRowVector(
+      ROW({"k1", "k2", "a", "b"}, {BIGINT(), BIGINT(), BIGINT(), VARCHAR()}),
+      10);
+
+  VELOX_ASSERT_USER_THROW(
+      PlanBuilder().values({data}).expand(
+          {{"k1", "k1", "a", "b", "0 as group_id_0"},
+           {"k1", "null", "a", "b", "1"},
+           {"null", "null", "a", "b", "2"}}),
+      fmt::format("Found duplicate column name in Expand plan node: k1."));
```

Comment:
fmt::format is redundant here and in the next statement. Please, remove from both places.

Reply:
@mbasmanova Removed.